test/ability_defiant_competitive.c

```diff
@@ -0,0 +1,152 @@
+#include "global.h"
+#include "test_battle.h"
+
+SINGLE_BATTLE_TEST("Status moves activate Defiant")
+{
+    PARAMETRIZE {}
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_CHARM].effect == EFFECT_ATTACK_DOWN_2);
+        PLAYER(SPECIES_WIGGLYTUFF) { Ability(ABILITY_COMPETITIVE); };
+        OPPONENT(SPECIES_PRIMEAPE) { Ability(ABILITY_DEFIANT); };
+    } WHEN {
+        TURN { MOVE(player, MOVE_CHARM); }
+    } SCENE {
+        ABILITY_POPUP(opponent, ABILITY_DEFIANT);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, opponent);
+        MESSAGE("Foe Primeape's Defiant sharply raised its Attack!");
+    } FINALLY {
+        EXPECT_EQ(opponent->statStages[STAT_ATK], DEFAULT_STAT_STAGE);
```

Comment:
Somebody else brought this up on Discord, but it would be preferable (and consistent with the existing tests) if all these checks against `statStages` were instead done by observing a change in the damage dealt (this is normally done by `PARAMETRIZING` over whether the ability triggers or not).

Some other minor test style notes:
- `PARAMETRIZE {}` doesn't do anything, so you wouldn't have to include it.
- I don't think `ABILITY_COMPETITIVE` is being used for anything? If that's true, then it shouldn't be specified.
- We typically use `SPECIES_WOBBUFFET` for Pokémon where the species doesn't matter (i.e. Wigglytuff in this example).
- You don't need the `;`s after `}`s in `PLAYER` and `OPPONENT`.